plugins/chat/app/services/chat/lookup_channel_threads.rb

```diff
@@ -87,25 +146,10 @@ def fetch_threads(guardian:, channel:, **)
           "original_messages.deleted_at IS NULL AND chat_messages.deleted_at IS NULL AND original_messages.id IS NOT NULL",
         )
         .group("chat_threads.id")
-        .order("MAX(chat_messages.created_at) DESC")
-        .limit(50)
-    end
-
-    def fetch_tracking(guardian:, threads:, **)
-      context.tracking =
-        ::Chat::TrackingStateReportQuery.call(
-          guardian: guardian,
-          thread_ids: threads.map(&:id),
-          include_threads: true,
-        ).thread_tracking
-    end
-
-    def fetch_memberships(guardian:, threads:, **)
-      context.memberships =
-        ::Chat::UserChatThreadMembership.where(
-          thread_id: threads.map(&:id),
-          user_id: guardian.user.id,
+        .select(
+          "chat_threads.id AS thread_id, MAX(chat_messages.created_at) AS latest_message_created_at, MAX(chat_messages.id) AS latest_message_id",
```

Comment:
Just out of curiosity, I wonder why we don't have some sort of `ChatThread#last_message_id` column instead of having to fetch this manually every time.

Reply:
This is a good point and could simplify querying in a lot of other places too, since we often have to show the last reply for threads. It shouldn't be too hard to keep in sync either because we have message creation / trashing services where we could do it. Will open up a TODO to do this in another PR